config/backend-config/internal/cache/cache.go

```diff
@@ -36,35 +36,35 @@ type cacheStore struct {
 // key is the key to use to store and fetch the config from the cache store
 //
 // ch is the channel to listen on for config updates and store them
-func Start(ctx context.Context, secret [32]byte, key string, ch pubsub.DataChannel) Cache {
+func Start(ctx context.Context, secret [32]byte, key string, ch pubsub.DataChannel) (Cache, error) {
```

Comment:
We may only subscribe if a connection has been acquired, so in line 69:
```
ch := channelProvider()
```

```suggestion
func Start(ctx context.Context, secret [32]byte, key string, channelProvider func() pubsub.DataChannel) (Cache, error) {
```